extract.js

```diff
@@ -196,10 +197,11 @@ Extract.prototype.destroy = function (err) {
   if (this._stream) this._stream.emit('close')
 }
 
-Extract.prototype._parse = function (size, onparse) {
+Extract.prototype._parse = function (size, onparse, newentry) {
   if (this._destroyed) return
   this._offset += size
   this._missing = size
+  if (newentry) this._partial = false
```

Comment:
Couldn't you just do `this._partial = onparse !== onheader` and not have the extra arg?